rust/src/libnm_dbus/connection/bridge.rs

```diff
@@ -16,51 +16,76 @@
 use std::collections::HashMap;
 use std::convert::TryFrom;
 
-use crate::{dbus_value::value_hash_get_bool, error::NmError};
+use crate::{dbus_value::own_value_to_bool, error::NmError};
 
 #[derive(Debug, Clone, PartialEq, Default)]
 pub struct NmSettingBridge {
     pub stp: Option<bool>,
+    _other: HashMap<String, zvariant::OwnedValue>,
 }
 
 impl TryFrom<&HashMap<String, zvariant::OwnedValue>> for NmSettingBridge {
     type Error = NmError;
     fn try_from(
-        value: &HashMap<String, zvariant::OwnedValue>,
+        setting_value: &HashMap<String, zvariant::OwnedValue>,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            stp: value_hash_get_bool(value, "stp")?,
-        })
+        let mut setting = Self::new();
+        for (key, value) in setting_value.iter() {
+            match key.as_str() {
+                "stp" => {
+                    setting.stp = Some(own_value_to_bool(value)?);
+                }
+                _ => {
+                    setting._other.insert(key.to_string(), value.clone());
+                }
+            }
+        }
```

Comment:
It could be shortened to something like:
```
let mut setting = Self::new();
let mut setting_cloned = setting_value.clone();
if let Some(stp_val) = setting_cloned.remove("stp") {
    setting.stp = Some(own_value_to_bool(&stp_val)?);
}
setting._other = setting_cloned;
```

I am not sure if that's desirable.

Maybe it would be worth to have some similar  helper to removed `value_hash_get_bool`. 
E.g. `value_hash_remove_bool` that would simplify this whole thing into:

```
let mut setting = Self::new();
let mut setting_cloned = setting_value.clone();
setting.stp = value_hash_remove_bool(&setting_cloned, "stp")?;
setting._other = setting_cloned;
```

Which is more readable IMO.

Reply:
In rust, I intend to refrain myself from using `clone()` whenever possible. Cloning the whole setting seems against this.

Reply:
That a valid point but the whole map is cloned anyway, in this case not cloning everything spares probably only the stp string from being cloned. Also another point might be that by cloning everything there is a single allocation happening instead of increasing it during insert (That could be avoided by pre-allocating the _other).

Reply:
If we do clone there, we will do cloning again when `own_value_to_string()` and others.

Reply:
That could be probably converted to something like:
```
pub(crate) fn own_value_to_string(
    value: zvariant::OwnedValue,
) -> Result<String, NmError> {
    check_value_is_string(&value)?;
    String::try_from(value).map_err(|e| {
        NmError::new(
            ErrorKind::Bug,
            format!("Failed to convert {:?} to string: {}", &value, e),
        )
    })
}
```

Yeah I know that cloning can be endless fight in rust

Reply:
New patch uploaded using this big clone() and extract memory from OwnValue out.